fastlane/lib/util.rb

```diff
@@ -45,3 +45,86 @@ def travis?
 def circle?
   ENV['CIRCLECI'] == 'true'
 end
+
+# diff two hashes at a path and return the changed items
+def deps_diff(old_hash, new_hash, path)
+  (new_hash.dig(*path).to_a - old_hash.dig(*path).to_a).to_h
+end
+
+def npm_native_package_changed?
+  old_package = JSON.parse(sh "git show '#{source_branch}:package.json'")
+  new_package = JSON.parse(sh "git show '#{current_branch}:package.json'")
+
+  changed_dependencies = deps_diff(old_package, new_package, ['dependencies']).keys
+  changed_devdependencies = deps_diff(old_package, new_package, ['devDependencies']).keys
+
+  dep_name_regex = /react|jsc/
+  (changed_dependencies + changed_devdependencies).any? { |dep| dep =~ dep_name_regex }
+end
+
+def native_build_globs
+  base_globs = [
+    '.circleci/**',
+    'e2e/**',
+    'fastlane/**',
+    'scripts/**',
+    'Gemfile.lock',
+  ]
+
+  case lane_context[:PLATFORM_NAME]
+  when :android
+    base_globs + [
+      'android/**',
+      '{modules,source}/**/*.{java,gradle}',
+    ]
+  when :ios
+    base_globs + [
+      'ios/**',
+      '{modules,source}/**/*.{h,m,mm}',
+    ]
+  else
+    base_globs
+  end
+end
+
+def native_file_changed?(changed_files)
+  native_build_globs.any? do |glob|
+    changed_files.any? do |path|
+      File.fnmatch?(glob, path)
+    end
+  end
+end
+
+# checks if the native build needs to be run
+def should_build?
+  # Essentially, we want to avoid native builds if the only thing that changed was JS code.
+
+  # TODO: figure out how to determine the actual source branch
+  source_branch = 'master'
+  current_branch = git_branch
+
+  if source_branch == current_branch
+    # if we're on master, we should do the build
+    UI.message("should_build? branch == #{source_branch}, so yes")
+    return true
+  end
+
+  # 1. need to get files changed in the current build since master
+  changed_files = sh("git log --name-only --format='' '#{source_branch}'..'#{current_branch}'").lines.sort.uniq
+
+  # 2. check for "packages we care about"
+  if changed_files.include? 'package.json' and npm_native_package_changed?
+    UI.message('should_build? some react|jsc dep changed, so yes')
```

Comment:
I ask about extracting `DEP_NAME_REGEXP` so that we can include that here:

Once you have it extracted, this should work:

```suggestion
    UI.message("should_build? some dependency matching #{DEP_NAME_REGEXP.inspect} changed, so yes")
```

Reply:
yeah just wasn't sure how to include it in the string before